libraries/shared/logs/extractor.go

```diff
@@ -45,8 +48,21 @@ type LogExtractor struct {
 	Fetcher                  fetcher.ILogFetcher
 	LogRepository            datastore.EventLogRepository
 	StartingBlock            *int64
+	EndingBlock              *int64
 	Syncer                   transactions.ITransactionsSyncer
 	Topics                   []common.Hash
+	RecheckHeaderCap         int64
+}
+
+func NewLogExtractor(db *postgres.DB, bc core.BlockChain) *LogExtractor {
+	return &LogExtractor{
+		CheckedHeadersRepository: repositories.NewCheckedHeadersRepository(db),
+		CheckedLogsRepository:    repositories.NewCheckedLogsRepository(db),
+		Fetcher:                  fetcher.NewLogFetcher(bc),
+		LogRepository:            repositories.NewEventLogRepository(db),
+		Syncer:                   transactions.NewTransactionsSyncer(db, bc),
+		RecheckHeaderCap:         constants.RecheckHeaderCap,
```

Comment:
👍 to moving this to assignment in the initializer - would be cool to be able to configure it via the CLI

Reply:
Yeah, I feel like being able to configure this and the starting/ending blocks via CLI could potentially make the `execute` more flexible. Esp for development.

Reply:
Yeah definitely. I think that being able to se this, and the Starting and Ending blocks via the CLI will help make it more flexible in the future. Especially for development. 🚀